### PR TITLE
receiver/prometheus/internal: direct metricfamily Prometheus->OTLP + equivalent tests

### PR DESCRIPTION
**Pull request by odeke-em**
Adds direct metricfamily Prometheus->OTLP conversion as well as equivalence tests.
After this change, there will be 2 other changes coming up and then finally we'll be able
to remove the dependency of OpenCensus in the Prometheus receiver.

Updates #3137
Follow-up of PR #3139 unfortunately given that Github doesn't support chaining PRs.

### CODE REVIEW

File: receiver/prometheusreceiver/internal/otlp_metricfamily.go
Comment (dashpole) [inline receiver/prometheusreceiver/internal/otlp_metricfamily.go:29]: might be useful to comment here that we are overriding fields in metricFamily, since that has implications beyond what is seen in this file if i'm reading it correctly.  Same for metricGroupPdata below

**Review by dashpole — APPROVED** [review]
lgtm with nits

File: receiver/prometheusreceiver/internal/otlp_metricsbuilder_test.go
Comment (bogdandrutu) [inline receiver/prometheusreceiver/internal/otlp_metricsbuilder_test.go:29]: Please add unittests for this functionality, otherwise we will be test-less when this gets removed.

**Review by bogdandrutu — COMMENTED** [review]
@rakyll @Aneurysm9 @dashpole as Prometheus experts please review.

File: receiver/prometheusreceiver/internal/otlp_metricsbuilder_test.go
Comment (odeke-em) [reply receiver/prometheusreceiver/internal/otlp_metricsbuilder_test.go:29]: Sounds good, thanks! Sending it shortly.

File: receiver/prometheusreceiver/internal/otlp_metricfamily.go
Comment (odeke-em) [reply receiver/prometheusreceiver/internal/otlp_metricfamily.go:32]: Done, thanks!

File: receiver/prometheusreceiver/internal/otlp_metricfamily.go
Comment (odeke-em) [reply receiver/prometheusreceiver/internal/otlp_metricfamily.go]: Done, thanks!

File: receiver/prometheusreceiver/internal/otlp_metricsbuilder_test.go
Comment (odeke-em) [reply receiver/prometheusreceiver/internal/otlp_metricsbuilder_test.go:31]: Done, please take a look. I've added unit tests for all the functions added in here. Thank you.

**Comment by odeke-em** [comment]
@bogdandrutu @dashpole please take a look again, unit tests added, but please remember that because Github can't stack PRs this change is waiting on PR #3139.

File: receiver/prometheusreceiver/internal/otlp_metricsbuilder.go
Comment (dashpole) [inline receiver/prometheusreceiver/internal/otlp_metricsbuilder.go:30]: this doesn't seem to be used anywhere.

Comment (odeke-em) [reply receiver/prometheusreceiver/internal/otlp_metricsbuilder.go]: Removed, thanks!

**Review by dashpole — APPROVED** [review]

**Comment by odeke-em** [comment]
@bogdandrutu, Anurag and David approved the PR, I addressed all the requested changes 7+ days ago; kindly waiting for your review/merge otherwise the PR has now been marked as stale.

**Comment by bogdandrutu** [comment]
Please fix lint errors

**Review by Aneurysm9 — APPROVED** [review]

**Comment by bogdandrutu** [comment]
@odeke-em ping on the lint errors

**Comment by odeke-em** [comment]
golangci-lint errors are unclear/phantom and pedantic -- max of 3 import groups which seems arbitrary and no way a Go convention, and it wasn't clear. However, I've addressed them. Thanks for the patience!

**Comment by tigrannajaryan** [comment]
@odeke-em thanks a lot for working on this! Can you please update the PR description / commit message to reflect how much of the refactoring this does and what remains to be done.

**Review by tigrannajaryan — APPROVED** [review]
Will merge once the PR descr/commit message are updated.

**Comment by odeke-em** [comment]
@tigrannajaryan I've updated the PR description. The PR is composed of multiple commits so updating the commit message would require me squashing them directly into one then force pushing which would lose context. I believe perhaps that when you go to squash+merge, you'll be able to copy+paste the updated description. Thank you for the review!

**Comment by tigrannajaryan** [comment]
Thanks @odeke-em 
I used the PR description as the commit message.

**Comment by odeke-em** [comment]
Awesome, thank you @tigrannajaryan!

**Pull request merged**
